### ISSUE TITLE
360 Stereo Capture when using HDRP

### ISSUE THREAD

**Issue by AndrewRH** [RenderHeads/UnityPlugin-AVProMovieCapture#37]
Unity seem to have left out 360 Stereo capture support in HDRP.....We'll wait for them to add support for it....

**Comment by eshaftel** [comment]
@AndrewRH -- Has anything changed here in the last two months on the Unity side? Trying to get 360 Stereo capture using URP and having the same issue of Top and Bottom being identical regardless of the inter-eye setting.  (except for particles appear to show depth, oddly).  Thinking about trying the Unity Recorder instead of AVPro Movie Capture, but the link suggests that it will run into the same fundamental problem, right?

**Comment by adriansotov** [comment]
Im having issues capturing monoscopic 360 video in URPL, anyone has managed to do it?

**Comment by AndrewRH** [comment]
@adriansotov 
Could you explain the issue is?

**Comment by adriansotov** [comment]
When using the AVPro Capture plugin, even the demo scene on Unity 2019.4.0f1 URPL this is how 360 video is being captured: 

![Annotation 2020-07-07 010946](https://user-images.githubusercontent.com/12679635/86853903-ada17380-c074-11ea-9ef4-6d34991e19f3.png)

If I follow the same procedure but without the URPL version of Unity everything works fine.

The video being captured is Monoscopic 360 video

**Comment by adriansotov** [comment]
This is how the capture looks in Unity without URLP, same exact process:
![Annotation2](https://user-images.githubusercontent.com/12679635/86854615-f73e8e00-c075-11ea-9c55-04b86b73fe86.png)

**Comment by AndrewRH** [comment]
@adriansotov i've tried but I can't replicate the issue you're having.  Can you replicate it using the demo scene "Demo04-360Capture.unity"?  

Could you tell us exactly how you're doing the capture - show us the settings etc.

Have you changed any settings in URP?

Thanks,

**Comment by adriansotov** [comment]
Ok @AndrewRH , I found the source of the issue.  If you capture with the headset connected to Unity you get the error, look:
The difference between the two captures is that in the bad one the Oculus Quest is connected to Unity.

![good](https://user-images.githubusercontent.com/12679635/86961401-666fbd00-c11e-11ea-8f09-1f2a92d273b2.png)

![Bad](https://user-images.githubusercontent.com/12679635/86961417-6e2f6180-c11e-11ea-997d-899708285ca1.png)

**Comment by adriansotov** [comment]
@AndrewRH  Im using Unity 2019.4.0f1 with the Universal Render Pipeline template and installing through Unity's package manager the XR Plugin Management tool and adding the Oculus plugging.  Target platform PC.
PLEASE NOTE, this only happens with the Universal Render Pipeline template, I have no issues with the 3D template.

**Comment by AndrewRH** [comment]
> @AndrewRH Im using Unity 2019.4.0f1 with the Universal Render Pipeline template and installing through Unity's package manager the XR Plugin Management tool and adding the Oculus plugging. Target platform PC.
> PLEASE NOTE, this only happens with the Universal Render Pipeline template, I have no issues with the 3D template.

It sounds like a bug in Unity.  Perhaps you could try upgrading to the  latest version of Unity 2019.4.3 and see if there are any further updates for those packages you're using.
I don't have a Oculus Quest headset that I can plug in for testing, so if you can find a way to replicate this that doesn't require hardware please let me know.

Thanks,

**Comment by AnriEU** [comment]
Unity 2020.3.22f1 HDRP
Empty project only with "AVPro Movie Capture (Windows)" 4.7.4
Capture From Camera 360 (VR)
It wont work. Video is blurred

https://user-images.githubusercontent.com/37102945/142373342-2b1412a3-b38e-4450-9407-6ad6d57bee03.mp4

**Comment by kahnivore** [comment]
@AnriEU could you try turning off motion blur in the HDRP default settings? 

(Project Settings -> HDRP Default Settings -> Rendering)

**Comment by AnriEU** [comment]
@kahnivore 
Test with
1. motion blur is ON
2. motion blur is OFF = better
3. motion blur is ON but used script "Capture from camera" = perfect
So, turn OFF did not help

https://drive.google.com/file/d/1OfgjJYpsSw96rUTM7AfGDqN-xGZ615pQ/view?usp=sharing

**Comment by kahnivore** [comment]
Right, but you've found a solution via using Capture from Camera? 

The post-processing in HDRP seems to interfere with the screen capture.

**Comment by AnriEU** [comment]
"Capture from Camera" works out of the box.
"Capture from Screen" works out of the box too.

"Capture From Camera 360" no.
Turn off "post-processing in HDRP" did not help.

I need "Capture From Camera 360"
Can you fix it?

**Comment by kahnivore** [comment]
Oh I see, you mean the demo, sorry. 

How did you turn off post-processing? Did you change any other settings? 

Attaching two videos below, captures of the Demo04-360Capture. The first looks similar to yours - very blurry. This is out of the box. The second looks clear, and all I've changed is that I've turned off motion blur as in the screenshot below. 

Could you try this in a fresh HDRP project with AVPro Movie Capture imported?

![image](https://user-images.githubusercontent.com/44468274/143279263-25ec18b0-8a61-47a6-b4a4-0984e9c2d972.png)


https://user-images.githubusercontent.com/44468274/143279316-deff04b2-cba5-4294-a58f-19e68064d813.mp4




https://user-images.githubusercontent.com/44468274/143279365-489a726e-45f1-42cd-b49e-1681ac66b0ba.mp4

**Comment by AnriEU** [comment]
There was a different problem. I had to change a little bit your code and made a lot of small tweaks in different places. And huge amount of testing time. And, wow! It is working now! Thanks!

**Comment by Chris-RH** [comment]
Hi @AnriEU,
I know its been a while since you posted this, but I'd be very interested to know what changes you made to get it to work, if you can remember?
Kind regards,
Chris

**Comment by gradeeterna** [comment]
Hey, is it possible to render stereoscopic 360 video in HDRP/URP yet, or still waiting for Unity to support it? Thanks
